src/power_grid_model/validation/validation.py

```diff
@@ -328,6 +328,13 @@ def validate_required_values(
         "tap_max",
         "tap_size",
     ]
+
+    # Regulators
+    required["regulator"] = required["base"] + ["regulated_object", "status"]
+    required["transformer_tap_regulator"] = required["regulator"]
+    if calculation_type is None or calculation_type == CalculationType.power_flow:
```

Comment:
Just curious about the reason behind this `calculation_type is None` condition, how so?

Reply:
If user doesn't specifically mention `calculation_type`, we should check all data provided.